ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/BeaconBlockBody.java

```diff
@@ -66,6 +67,10 @@ default Optional<ExecutionPayloadHeader> getOptionalExecutionPayloadHeader() {
     return Optional.empty();
   }
 
+  default Optional<ExecutionPayloadSummary> getOptionalExecutionPayloadSummary() {
+    return Optional.empty();
```

Comment:
I'd suggest:
```suggestion
    return getOptionalExecutionPayload().or(getOptionalExecutionPayloadHeader());
```

That way if either of those are overridden this method just works (but still makes sense to explicitly override it as you've done as well to save checking both options).